components/shortcuts/shortcuts_modal/shortcuts_modal.tsx

```diff
@@ -0,0 +1,236 @@
+    },
+    info: {
+        id: t('shortcuts.info'),
+        defaultMessage:
+      'Begin a message with / for a list of all the commands at your disposal.',
+    },
+    navHeader: {
+        id: t('shortcuts.nav.header'),
+        defaultMessage: 'Navigation',
+    },
+});
+
+type Props = {
+    intl: IntlShape;
+}
+
+type State = {
+    show: boolean;
+}
+
+class ShortcutsModal extends React.PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            show: false,
+        };
+    }
+
+    componentDidMount() {
+        ModalStore.addModalListener(
+            Constants.ActionTypes.TOGGLE_SHORTCUTS_MODAL,
+            this.handleToggle,
+        );
+    }
+
+    componentWillUnmount() {
+        ModalStore.removeModalListener(
+            Constants.ActionTypes.TOGGLE_SHORTCUTS_MODAL,
+            this.handleToggle,
+        );
+    }
+
+    handleToggle = () => {
+        //toggles the state of shortcut dialog
+        this.setState({
+            show: !this.state.show,
+        });
+    };
+
+    handleHide = () => {
+        this.setState({show: false});
+    };
+
+    render() {
+        const {formatMessage} = this.props.intl;
+
+        const isLinux = Utils.isLinux();
+
+        return (
+            <Modal
+                dialogClassName='a11y__modal shortcuts-modal'
+                show={this.state.show}
+                onHide={this.handleHide}
+                onExited={this.handleHide}
+                role='dialog'
+                aria-labelledby='shortcutsModalLabel'
+            >
+                <div className='shortcuts-content'>
+                    <Modal.Header closeButton={true}>
+                        <Modal.Title
+                            componentClass='h1'
+                            id='shortcutsModalLabel'
+                        >
+                            <strong>
+                                <ShortcutSequence shortcut={shortcuts.mainHeader}/>
+                            </strong>
+                        </Modal.Title>
+                    </Modal.Header>
+                    <Modal.Body>
+                        <div className='row'>
+                            <div className='col-sm-4'>
+                                <div className='section'>
+                                    <div>
+                                        <h4 className='section-title'>
+                                            <strong>
+                                                {formatMessage({
+                                                    id: t('shortcuts.nav.header'),
+                                                    defaultMessage: 'Navigation',
+                                                })}
+                                            </strong>
+                                        </h4>
+                                        <ShortcutSequence shortcut={shortcuts.navPrev}/>
```

Comment:
We can do that, but the items in the array are kept without reference to the subcategory in the modal, so not sure how it can be done without rewriting the shortcuts to contain that information.